chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -128,10 +128,22 @@ pub struct TransactionVerifierDelta {
     token_issuance_cache: ConsumedTokenIssuanceCache,
 }
 
+#[derive(Clone)]
+pub struct TransactionVerifierConfig {
+    pub tx_index_enabled: bool,
+}
+
+impl TransactionVerifierConfig {
+    pub fn new(tx_index_enabled: bool) -> Self {
```

Comment:
It would be nice to have a custom enum for the parameter (and the struct field), like
```rust
enum TxIndexEnabled { Yes, No }
```

It's difficult to see what the parameter refers to if it's just `TransactionVerifierConfig::new(true)`. You can implement `From<bool>` for it if you like

Reply:
We have multiple issues with this whole thing... I'll come back to this problem later.

Reply:
See this for more details: https://github.com/mintlayer/mintlayer-core/pull/542#discussion_r1017612540